unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/EntityInformation/AttributeXYZ.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+
+public class AttributeXYZ : MonoBehaviour
+{
+    public TMP_InputField xField;
+    public TMP_InputField yField;
+    public TMP_InputField zField;
+
+    public event Action<Vector3> OnChanged;
+
+    Vector3 currentValue;
+
+    bool isSelected = false;
+    
+    public void SetValues(Vector3 value)
+    {
+        if (!isSelected)
+        {
+            currentValue = value;
+            xField.SetTextWithoutNotify(value.x.ToString("0.##"));
+            yField.SetTextWithoutNotify(value.y.ToString("0.##"));
+            zField.SetTextWithoutNotify(value.z.ToString("0.##"));
+        }
+    }
+
+    public void ChangeXValue(string value)
+    {
+        if (!isSelected)
+            return;
+
+        currentValue.x = float.Parse(value);
+        OnChanged?.Invoke(currentValue);
+    }
+
+    public void ChangeYValue(string value)
+    {
+        if (!isSelected)
+            return;
+
+        currentValue.y = float.Parse(value);
+        OnChanged?.Invoke(currentValue);
+    }
+
+    public void ChangeZValue(string value)
+    {
+        if (!isSelected)
+            return;
+
+        currentValue.z = float.Parse(value);
+        OnChanged?.Invoke(currentValue);
+    }
```

Comment:
These methods are used somewhere? I don't see any reason to keep them if we are just going to use the more comfy `SetValues` method.

Reply:
Yes they are used from the Unity Inspector - `TMP_InputField` `ChangeValue ` They are called every time that a user made a change on the input so we can update the value accordly, same for the `InputSelected `and `InputDeselected`

Reply:
Can we have these events explicitly subscribed in the code then? Linking them in Unity makes tracking references really difficult.

Reply:
Sure! However, this will limit the way a designer can assign that kind of behavior